### PR TITLE
hadron-security package

### PR DESCRIPTION
**Pull request by Miczeq22**

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -8,11 +8,13 @@
...
     "packages/hadron-json-provider",
     "packages/hadron-serialization",
+    "packages/hadron-security",
     "packages/hadron-validation",
     "packages/hadron-typeorm",
     "packages/hadron-events",
     "packages/hadron-express",
-    "packages/hadron-demo"
+    "packages/hadron-demo",
+    "packages/hadron-security"
```
Comment (szymonbrainhub) [inline lerna.json:16]: hadron-security added twice

File: package.json [hunk 2]
```diff
@@ -25,16 +25,32 @@
   "author": "Brainhub",
   "license": "MIT",
   "dependencies": {
+    "@types/bcrypt": "1.0.0",
```
Comment (szymonbrainhub) [inline package.json]: types should be in devDependencies

File: package.json [hunk 3]
```diff
@@ -25,16 +25,32 @@
...
     "@types/bunyan": "1.8.4",
+    "bcrypt": "2.0.0",
```
Comment (szymonbrainhub) [inline package.json]: Such dependencies should be defined in package.json of specific package. Same with `jsonwebtoken`

File: packages/hadron-demo/index.ts [hunk 4]
```diff
@@ -5,41 +5,66 @@ import * as hadronEvents from '@brainhubeu/hadron-events';
...
       ...serializationRoutes,
       ...routes,
+      ...expressConfig.routes,
     },
   };
 
   hadron(
     expressApp,
-    [hadronEvents, hadronSerialization, hadronExpress, hadronTypeORM],
+    [hadronEvents, hadronSerialization, hadronTypeOrm],
```
Comment (szymonbrainhub) [inline packages/hadron-demo/index.ts:42]: to be discussed

File: packages/hadron-security/index.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import IRoleProvider from './src/IRoleProvider';
+import { IUser, IRole } from './src/hierarchyProvider';
+import InMemoryRoleProvider from './src/in-memory/InMemoryRoleProvider';
+import InMemoryUserProvider from './src/in-memory/InMemoryUserProvider';
+import expressMiddlewareProvider, {
+  generateTokenMiddleware,
+} from './src/providers/expressProvider';
+import * as bcrypt from './src/password/bcrypt/bcrypt';
+
+export const register = (container: any, config: any) => {
```
Comment (szymonbrainhub) [inline packages/hadron-security/index.ts:12]: if register is not needed,  You can safely remove it from here

**Review by szymonbrainhub — COMMENTED** [review]
Let's discuss this PR. There is way too many changes for one PR and it will be easier to review that peer-to-peer

**Comment by damian66** [comment]
@szymonbrainhub @Miczeq22 any update to this PR ?

**Comment by szymonbrainhub** [comment]
Work in progress. We need to figure out better initialization than the current one ( https://github.com/brainhubeu/hadron/pull/4/files#diff-89ea6071a9cbbf8039d61c4e007b8715R44 )

**Comment by damian-szulc** [comment]
note that the name `hadron-security` can be misled with a package providing some security improvements (for example anti cross-site-request-forgery). Please consider renaming it `hadron-auth`.

**Comment by szymonbrainhub** [comment]
further requirements: 
- merge security configuration with express configuration
- fix tests
- better initialization (with register function)

**Pull request closed without merging**